homeassistant/components/flux_led/light.py

```diff
@@ -315,10 +317,24 @@ def update(self):
         else:
             self._mode = MODE_RGB
 
-        self._white_value = self._get_rgbw[3]
+        if self._mode == MODE_RGBCW:
+            white_temp = self.temperature_cw
+            self._white_value = (
+                white_temp[0] if white_temp[0] > white_temp[1] else white_temp[1]
+            )
+        elif self._mode == MODE_RGBWW:
+            self._white_value = self.temperature_ww
+        else:
+            self._white_value = self._get_rgbw[3]
 
         if self._mode == MODE_WHITE:
             self._brightness = self._white_value
+        elif self._mode == MODE_RGBCW:
+            rgbww = self.temperature_cw
```

Comment:
We should be ild cache this call in a local variable so we don't need to make it twice in the same update.

Also it looks like our if/elif logic isn't optimal. We're check the same mode more than once. Please optimize this.